bug(eslint-plugin): signals configuration

Sorry to bother you again, but I have another problem, if I now extend from `plugin:@ngrx/all`, I have the following errors:

```sh
  1:1  error  Definition for rule '@ngrx/signal-state-no-arrays-at-root-level' was not found  @ngrx/signal-state-no-arrays-at-root-level
  1:1  error  Definition for rule '@ngrx/with-state-no-arrays-at-root-level' was not found    @ngrx/with-state-no-arrays-at-root-level
```

If I turn these rules off, it's working again, am I missing something?

_Originally posted by @Julien-Marcou in https://github.com/ngrx/platform/issues/4381#issuecomment-2158864774_

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -26,7 +26,6 @@
       "files": ["*.ts", "*.tsx"],
       "extends": ["plugin:@nx/typescript"],
       "rules": {
-        "@typescript-eslint/no-extra-semi": "error",
         "no-extra-semi": "off"
       }
     },
@@ -34,7 +33,6 @@
       "files": ["*.js", "*.jsx"],
       "extends": ["plugin:@nx/javascript"],
       "rules": {
-        "@typescript-eslint/no-extra-semi": "error",
         "no-extra-semi": "off"
       }
     },
@@ -54,6 +52,7 @@
         "@typescript-eslint/prefer-namespace-keyword": "error",
         "@typescript-eslint/ban-types": "warn",
         "@typescript-eslint/no-empty-function": "warn",
+        "@typescript-eslint/no-empty-object-type": "warn",
         "eqeqeq": ["off", "smart"],
         "id-blacklist": [
           "error",

--- a/modules/data/src/utils/default-logger.ts
+++ b/modules/data/src/utils/default-logger.ts
@@ -5,19 +5,31 @@ import { Logger } from './interfaces';
 export class DefaultLogger implements Logger {
   error(message?: any, extra?: any) {
     if (message) {
-      extra ? console.error(message, extra) : console.error(message);
+      if (extra) {
+        console.error(message, extra);
+      } else {
+        console.error(message);
+      }
     }
   }
 
   log(message?: any, extra?: any) {
     if (message) {
-      extra ? console.log(message, extra) : console.log(message);
+      if (extra) {
+        console.log(message, extra);
+      } else {
+        console.log(message);
+      }
     }
   }
 
   warn(message?: any, extra?: any) {
     if (message) {
-      extra ? console.warn(message, extra) : console.warn(message);
+      if (extra) {
+        console.warn(message, extra);
+      } else {
+        console.warn(message);
+      }
     }
   }
 }

--- a/modules/entity/spec/fixtures/book.ts
+++ b/modules/entity/spec/fixtures/book.ts
@@ -1,4 +1,4 @@
-// eslint-disable-next-line @typescript-eslint/no-var-requires
+// eslint-disable-next-line @typescript-eslint/no-var-requires,@typescript-eslint/no-require-imports
 const deepFreeze = require('deep-freeze');
 
 export interface BookModel {

--- a/modules/eslint-plugin/package.json
+++ b/modules/eslint-plugin/package.json
@@ -45,6 +45,7 @@
   },
   "peerDependencies": {
     "eslint": "^8.57.0 || ^9.0.0",
+    "@typescript-eslint/utils": "^7.11.0 || ^8.0.0-alpha.20",
     "typescript": "*"
   }
 }

--- a/modules/eslint-plugin/scripts/generate-config.ts
+++ b/modules/eslint-plugin/scripts/generate-config.ts
@@ -39,7 +39,7 @@ function writeConfig(
     },
     {}
   );
-  const parserOptions: null | Record<string, string | number> =
+  const requireParserOptions: null | Record<string, string | number> =
     rulesForConfig.some(([_, rule]) => rule.meta.docs?.requiresTypeChecking)
       ? {
           ecmaVersion: 2020,
@@ -75,8 +75,12 @@ function writeConfig(
         languageOptions: {
           parser,
           ${
-            parserOptions
-              ? `parserOptions: ${JSON.stringify(parserOptions, null, 2)},`
+            requireParserOptions
+              ? `parserOptions: ${JSON.stringify(
+                  requireParserOptions,
+                  null,
+                  2
+                )},`
               : ''
           }
         },
@@ -94,7 +98,7 @@ function writeConfig(
     plugins: ['@ngrx'],
     rules: configRules,
   };
-  if (configName.includes('all')) {
+  if (requireParserOptions) {
     jsonConfig.parserOptions = {
       ecmaVersion: 2020,
       sourceType: 'module',

--- a/modules/eslint-plugin/spec/exported-rules.spec.ts
+++ b/modules/eslint-plugin/spec/exported-rules.spec.ts
@@ -1,0 +1,39 @@
+import * as path from 'path';
+import * as lib from '../src/rules';
+import { traverseFolder } from '../src/utils';
+import { configs } from '../v9';
+
+const rulesDirectory = path.join(__dirname, '../src/rules');
+const configsDirectory = path.join(__dirname, '../src/configs');
+
+function getAllRules() {
+  return [...traverseFolder(rulesDirectory, ['.ts'])]
+    .map((rule) => rule.file)
+    .filter((rule) => rule !== 'index');
+}
+function getAllConfigs() {
+  return [...traverseFolder(configsDirectory, ['.ts'])];
+}
+
+describe('ESLint V8', () => {
+  test('exports all rules', () => {
+    const rules = getAllRules();
+    expect(Object.keys(lib.rules).length).toBe(rules.length);
+  });
+  test('exports all configurations', () => {
+    const configs = getAllConfigs();
+    expect(configs.length).toBe(6);
+  });
+});
+
+describe('ESLint V9', () => {
+  test('exports all rules ', () => {
+    const rules = getAllRules();
+    expect(Object.keys((configs.all[1] as any).rules).length).toBe(
+      rules.length
+    );
+  });
+  test('exports all configurations', () => {
+    expect(Object.keys(configs).length).toBe(6);
+  });
+});

--- a/modules/eslint-plugin/src/configs/effects.json
+++ b/modules/eslint-plugin/src/configs/effects.json
@@ -9,5 +9,10 @@
     "@ngrx/prefer-action-creator-in-of-type": "error",
     "@ngrx/prefer-effect-callback-in-block-statement": "error",
     "@ngrx/use-effects-lifecycle-interface": "error"
+  },
+  "parserOptions": {
+    "ecmaVersion": 2020,
+    "sourceType": "module",
+    "project": "./tsconfig.json"
   }
 }

--- a/modules/eslint-plugin/src/configs/signals.json
+++ b/modules/eslint-plugin/src/configs/signals.json
@@ -4,5 +4,10 @@
   "rules": {
     "@ngrx/signal-state-no-arrays-at-root-level": "error",
     "@ngrx/with-state-no-arrays-at-root-level": "error"
+  },
+  "parserOptions": {
+    "ecmaVersion": 2020,
+    "sourceType": "module",
+    "project": "./tsconfig.json"
   }
 }

--- a/modules/eslint-plugin/src/rule-creator.ts
+++ b/modules/eslint-plugin/src/rule-creator.ts
@@ -14,7 +14,7 @@ type CreateRuleMeta<
   TMessageIds extends string,
   TOptions extends readonly unknown[]
 > = {
-  docs: Omit<TSESLint.RuleMetaDataDocs<TOptions>, 'url'> & {
+  docs: Omit<TSESLint.RuleMetaDataDocs, 'url'> & {
     requiresTypeChecking?: boolean;
   };
 } & Omit<Meta<TMessageIds, TOptions>, 'docs'>;

--- a/modules/eslint-plugin/src/rules/index.ts
+++ b/modules/eslint-plugin/src/rules/index.ts
@@ -31,6 +31,8 @@ import selectStyle from './store/select-style';
 import useConsistentGlobalStoreName from './store/use-consistent-global-store-name';
 // operators
 import preferConcatLatestFrom from './operators/prefer-concat-latest-from';
+import signalStateNoArraysAtRootLevel from './signals/signal-state-no-arrays-at-root-level';
+import withStateNoArraysAtRootLevel from './signals/with-state-no-arrays-at-root-level';
 
 export const rules = {
   // component-store
@@ -71,4 +73,7 @@ export const rules = {
   'use-consistent-global-store-name': useConsistentGlobalStoreName,
   // operators
   'prefer-concat-latest-from': preferConcatLatestFrom,
+  // signals
+  'signal-state-no-arrays-at-root-level': signalStateNoArraysAtRootLevel,
+  'with-state-no-arrays-at-root-level': withStateNoArraysAtRootLevel,
 };

--- a/modules/eslint-plugin/src/utils/helper-functions/rules.ts
+++ b/modules/eslint-plugin/src/utils/helper-functions/rules.ts
@@ -7,7 +7,7 @@ const interopRequireDefault = (obj: any): { default: unknown } =>
   obj && obj.__esModule ? obj : { default: obj };
 
 const importDefault = (moduleName: string) =>
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  // eslint-disable-next-line @typescript-eslint/no-var-requires,@typescript-eslint/no-require-imports
   interopRequireDefault(require(moduleName)).default;
 
 const rulesDir = path.join(__dirname, '../../rules');

--- a/modules/eslint-plugin/src/utils/helper-functions/versions.ts
+++ b/modules/eslint-plugin/src/utils/helper-functions/versions.ts
@@ -15,7 +15,7 @@ function getNgrxVersion(pkg: string): string {
 
 function readPlatformVersion(pkg: string) {
   try {
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    // eslint-disable-next-line @typescript-eslint/no-var-requires,@typescript-eslint/no-require-imports
     const ngrxVersion = require(`${pkg}/package.json`) as {
       version?: string;
     };

--- a/modules/schematics/src/store/index.spec.ts
+++ b/modules/schematics/src/store/index.spec.ts
@@ -79,7 +79,7 @@ describe('Store Schematic', () => {
     expect(tree.readContent(modulePath)).toMatchSnapshot();
 
     expect(tree.files.indexOf(reducersPath)).toBeGreaterThanOrEqual(0);
-    expect(tree.readContent(reducersPath)).toMatchSnapshot;
+    expect(tree.readContent(reducersPath)).toMatchSnapshot();
   });
 
   it('should create the initial store to specified project if provided', async () => {
@@ -132,7 +132,7 @@ describe('Store Schematic', () => {
     expect(tree.readContent(modulePath)).toMatchSnapshot();
 
     expect(tree.files.indexOf(reducersPath)).toBeGreaterThanOrEqual(0);
-    expect(tree.readContent(reducersPath)).toMatchSnapshot;
+    expect(tree.readContent(reducersPath)).toMatchSnapshot();
   });
 
   it('should fail if specified module does not exist', async () => {
@@ -153,7 +153,7 @@ describe('Store Schematic', () => {
     expect(tree.readContent(modulePath)).toMatchSnapshot();
 
     expect(tree.files.indexOf(reducersPath)).toBeGreaterThanOrEqual(0);
-    expect(tree.readContent(reducersPath)).toMatchSnapshot;
+    expect(tree.readContent(reducersPath)).toMatchSnapshot();
   });
 
   it('should support a default root state interface name', async () => {
@@ -167,7 +167,7 @@ describe('Store Schematic', () => {
     expect(tree.readContent(modulePath)).toMatchSnapshot();
 
     expect(tree.files.indexOf(reducersPath)).toBeGreaterThanOrEqual(0);
-    expect(tree.readContent(reducersPath)).toMatchSnapshot;
+    expect(tree.readContent(reducersPath)).toMatchSnapshot();
   });
 
   it('should support a custom root state interface name', async () => {
@@ -185,7 +185,7 @@ describe('Store Schematic', () => {
     expect(tree.readContent(modulePath)).toMatchSnapshot();
 
     expect(tree.files.indexOf(reducersPath)).toBeGreaterThanOrEqual(0);
-    expect(tree.readContent(reducersPath)).toMatchSnapshot;
+    expect(tree.readContent(reducersPath)).toMatchSnapshot();
   });
 
   it('should support a default feature state interface name', async () => {
@@ -199,7 +199,7 @@ describe('Store Schematic', () => {
     expect(tree.readContent(modulePath)).toMatchSnapshot();
 
     expect(tree.files.indexOf(reducersPath)).toBeGreaterThanOrEqual(0);
-    expect(tree.readContent(reducersPath)).toMatchSnapshot;
+    expect(tree.readContent(reducersPath)).toMatchSnapshot();
   });
 
   it('should support a custom feature state interface name', async () => {
@@ -218,7 +218,7 @@ describe('Store Schematic', () => {
     expect(tree.readContent(modulePath)).toMatchSnapshot();
 
     expect(tree.files.indexOf(reducersPath)).toBeGreaterThanOrEqual(0);
-    expect(tree.readContent(reducersPath)).toMatchSnapshot;
+    expect(tree.readContent(reducersPath)).toMatchSnapshot();
   });
 
   it('should fail if a feature state name is not specified', async () => {
@@ -255,7 +255,7 @@ describe('Store Schematic', () => {
     expect(tree.readContent(modulePath)).toMatchSnapshot();
 
     expect(tree.files.indexOf(reducersPath)).toBeGreaterThanOrEqual(0);
-    expect(tree.readContent(reducersPath)).toMatchSnapshot;
+    expect(tree.readContent(reducersPath)).toMatchSnapshot();
   });
 
   it('should not add a feature key if root', async () => {
@@ -269,7 +269,7 @@ describe('Store Schematic', () => {
     expect(tree.readContent(modulePath)).toMatchSnapshot();
 
     expect(tree.files.indexOf(reducersPath)).toBeGreaterThanOrEqual(0);
-    expect(tree.readContent(reducersPath)).toMatchSnapshot;
+    expect(tree.readContent(reducersPath)).toMatchSnapshot();
   });
 
   it('should add the initial config correctly into an empty module', async () => {

--- a/modules/store/migrations/8_0_0-rc/index.ts
+++ b/modules/store/migrations/8_0_0-rc/index.ts
@@ -15,7 +15,8 @@ import {
 function replaceWithRuntimeChecks(): Rule {
   return (tree: Tree) => {
     // only add runtime checks when ngrx-store-freeze is used
-    visitTSSourceFiles<boolean>(tree, removeUsages) &&
+    const _ =
+      visitTSSourceFiles<boolean>(tree, removeUsages) &&
       visitTSSourceFiles(tree, insertRuntimeChecks);
   };
 }

--- a/package.json
+++ b/package.json
@@ -124,9 +124,9 @@
     "@types/rimraf": "^0.0.28",
     "@types/semver": "^7.3.9",
     "@types/shelljs": "^0.8.5",
-    "@typescript-eslint/parser": "7.11.0",
-    "@typescript-eslint/rule-tester": "7.11.0",
-    "@typescript-eslint/utils": "7.11.0",
+    "@typescript-eslint/parser": "8.0.0-alpha.20",
+    "@typescript-eslint/rule-tester": "8.0.0-alpha.20",
+    "@typescript-eslint/utils": "8.0.0-alpha.20",
     "chokidar": "^3.5.3",
     "chokidar-cli": "^1.2.0",
     "conventional-changelog": "^1.1.4",
@@ -185,7 +185,7 @@
     "tsickle": "^0.37.0",
     "tsutils": "2.27.2",
     "typescript": "5.4.5",
-    "typescript-eslint": "7.11.0",
+    "typescript-eslint": "8.0.0-alpha.20",
     "uglify-js": "^3.1.9"
   },
   "collective": {

--- a/projects/ngrx.io/.eslintrc.json
+++ b/projects/ngrx.io/.eslintrc.json
@@ -43,7 +43,6 @@
             "accessibility": "explicit"
           }
         ],
-        "@typescript-eslint/indent": "error",
         "@typescript-eslint/member-delimiter-style": [
           "off",
           {
@@ -68,9 +67,7 @@
         ],
         "@typescript-eslint/no-unused-expressions": "error",
         "@typescript-eslint/prefer-function-type": "error",
-        "@typescript-eslint/quotes": ["error", "single"],
         "@typescript-eslint/semi": ["off", null],
-        "@typescript-eslint/type-annotation-spacing": "error",
         "@typescript-eslint/unified-signatures": "error",
         "@typescript-eslint/no-non-null-assertion": "off",
         "brace-style": ["error", "1tbs"],

--- a/projects/ngrx.io/content/guide/eslint-plugin/index.md
+++ b/projects/ngrx.io/content/guide/eslint-plugin/index.md
@@ -110,7 +110,13 @@ Instead of manually configuring the rules, there are also [preconfigured configu
 <!-- DO NOT EDIT, this table is automatically generated-->
 <!-- CONFIGURATIONS-CONFIG:START -->
 
-| Name |
-| ---- |
+| Name                                                                                                                 |
+| -------------------------------------------------------------------------------------------------------------------- |
+| [all](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/all.json)                         |
+| [component-store](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/component-store.json) |
+| [effects](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/effects.json)                 |
+| [operators](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/operators.json)             |
+| [signals](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/signals.json)                 |
+| [store](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/store.json)                     |
 
   <!-- CONFIGURATIONS-CONFIG:END -->

--- a/projects/ngrx.io/src/app/custom-elements/api/api-list.component.spec.ts
+++ b/projects/ngrx.io/src/app/custom-elements/api/api-list.component.spec.ts
@@ -241,7 +241,6 @@ class TestApiService {
     sections = this.sectionsSubject.asObservable();
 }
 
-/* eslint-disable @typescript-eslint/quotes */
 const apiSections: ApiSection[] = [
     {
         name: 'common',

--- a/projects/ngrx.io/src/app/custom-elements/resource/resource.service.spec.ts
+++ b/projects/ngrx.io/src/app/custom-elements/resource/resource.service.spec.ts
@@ -91,7 +91,6 @@ describe('ResourceService', () => {
 });
 
 function getTestResources() {
-    /* eslint-disable @typescript-eslint/quotes */
     return {
         'Cat 3': {
             order: 3,

